### PR TITLE
[Metricbeat] Convert HAProxy to reporter metricset

### PR DESCRIPTION
**Pull request by ruflin**
This is required to add proper ECS fields to HAProxy.

### CODE REVIEW

File: metricbeat/module/haproxy/info/data.go [hunk 1]
```diff
@@ -145,14 +145,16 @@ func eventMapping(info *haproxy.Info) (common.MapStr, error) {
 			// Convert this value to a float between 0.0 and 1.0
 			fval, err := strconv.ParseFloat(f.Interface().(string), 64)
```
Comment (sayden) [inline metricbeat/module/haproxy/info/data.go]: I was thinking to open an issue to _improve_ potential nil errors here. WDYT @ruflin?

File: metricbeat/module/haproxy/info/data.go [hunk 2]
```diff
@@ -145,14 +145,16 @@ func eventMapping(info *haproxy.Info) (common.MapStr, error) {
...
 			if err != nil {
-				return nil, err
+				r.Error(err)
```
Comment (sayden) [inline metricbeat/module/haproxy/info/data.go]: In the rest of the metricset you are wrapping errors and giving them more context. Not here?

Comment (ruflin) [reply metricbeat/module/haproxy/info/data.go]: This PR is keeping all the behaviour as is as it's required for ECS. Agree that we need to do some more cleanup and do better with error message. Perhaps we can add a general item about error handling here?

File: metricbeat/module/haproxy/info/data.go (hunk 1 above)
Comment (ruflin) [reply metricbeat/module/haproxy/info/data.go]: sure. If it's a quick fix, no need for an issue, directly to a PR ;-)

**Pull request merged**